src/types/decimal.rs

```diff
@@ -295,41 +295,37 @@ impl TryFrom<f64> for Decimal {
 }
 
 impl Display for Decimal {
+    #[rustfmt::skip] // https://github.com/rust-lang/rustfmt/issues/3255
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        // Inspired by the formatting conventions of Java's BigDecimal.toString()
+        const WIDE_NUMBER: i64 = 6; // if you think about it, six is a lot 🙃
```

Comment:
This could be a `usize`, which would allow you to skip:
* casting `len as i64` down on line 308
* casting `WIDE_NUMBER as usize` on line 314

Reply:
Yeah `WIDE_NUMBER` should be a `usize`.

* If I don't cast `len as i64` on line 308 then`dot_index` gets inferred as `usize`, but that's not right because `dot_index` should be able to be negative.
* As far as I understand it, I either have to cast `WIDE_NUMBER as usize` on line 314 to agree with `len` or cast `WIDE_NUMBER as i64` on line 324 for easiest comparison with the negative-valued `dot_index` there. The latter is the right call because `WIDE_NUMBER` *should* be a `usize`. 

Unless I'm missing something, which is probable :) I'd also happily take feedback or second opinions on the `WIDE_NUMBER` concept- whether we ought to  have it at all, whether it ought to be applied "evenly" on both whole and fractional numbers, and how large it should be.